src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,169 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private int capacity;
+    private int threshold;
+    private int size;
+    private double loadFactor;
+    private Node<K, V>[] table;
+    private final int defaultInitialCapacity = 16;
+    private final double defaultLoadFactor = 0.75f;
+
+    public MyHashMap() {
+        this.capacity = defaultInitialCapacity;
+        this.threshold = (int) (capacity * defaultLoadFactor);
+        this.loadFactor = defaultLoadFactor;
+        this.table = new Node[defaultInitialCapacity];
+    }
+
+    public class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
+
+        public int getHash() {
+            return hash;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public Node<K, V> getNext() {
+            return next;
+        }
+    }
+
+    public int getHash(Object key) {
+        int hash = 43;
+        return (key == null) ? 0 : (hash * 17 + Objects.hashCode(key) ^ hash);
+    }
+
+    public boolean isKeyEqual(K key, Node<K, V> currentNode) {
+        if (key == null && currentNode.getKey() == null) {
+            return true;
+        } else {
+            return currentNode.getKey() != null && currentNode.getKey().equals(key);
+        }
```

Comment:
combine checks. In this task, you are free to use Objects.equals btw

Reply:
Yes, ok